misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -177,11 +246,15 @@
                         if (matchedNode.parent().kind() == SyntaxKind.CHECK_EXPRESSION) {
                             matchedNode = matchedNode.parent();
                         }
+                    } else if (positionDetails.matchedNode().kind() == SyntaxKind.TYPE_CAST_EXPRESSION) {
+                        if (((TypeCastExpressionNode) positionDetails.matchedNode()).expression().kind() ==
+                                SyntaxKind.FUNCTION_CALL) {
+                            matchedNode = ((TypeCastExpressionNode) positionDetails.matchedNode()).expression();
+                        }
```

Comment:
We can merge these two conditions together

Reply:
Merged the two conditions